worker/src/DepLibUring.cpp

```diff
@@ -0,0 +1,411 @@
+#define MS_CLASS "DepLibUring"
+
+// #define MS_LOG_DEV_LEVEL 3
+
+#include "DepLibUring.hpp"
+#include "Logger.hpp"
+#include "MediaSoupErrors.hpp"
+#include <sys/eventfd.h>
+#include <sys/utsname.h>
+
+/* Static variables. */
+
+/* liburing instance per thread. */
+thread_local DepLibUring::LibUring* DepLibUring::liburing{ nullptr };
+/* Completion queue entry array used to retrieve processes tasks. */
+thread_local struct io_uring_cqe* cqes[DepLibUring::QueueDepth];
+
+/* Static methods for UV callbacks. */
+
+inline static void onCloseFd(uv_handle_t* handle)
+{
+	delete reinterpret_cast<uv_poll_t*>(handle);
+}
+
+inline static void onFdEvent(uv_poll_t* handle, int status, int events)
+{
+	auto* liburing = static_cast<DepLibUring::LibUring*>(handle->data);
+
+	auto count = io_uring_peek_batch_cqe(liburing->GetRing(), cqes, DepLibUring::QueueDepth);
+
+	// libuv uses level triggering, so we need to read from the socket to reset
+	// the counter in order to avoid libuv calling this callback indefinitely.
+	eventfd_t v;
+	int err = eventfd_read(liburing->GetEventFd(), std::addressof(v));
+	if (err < 0)
+	{
+		MS_ERROR("eventfd_read() failed: %s", std::strerror(-err));
```

Comment:
If this error happens then what? There is no return; here. Code below weeks running.

Reply:
If this error happens we should abort. This call should never fail.

Reply:
I'll modify it.

Reply:
> Good. Then change it to MS_ABORT()? ¯_(ツ)_/¯

yes yes..., of course.